instrumentation/http/src/main/java/brave/http/HttpServerRequest.java

```diff
@@ -39,13 +39,34 @@
   }
 
   /**
-   * Override and return true when it is possible to parse the {@link Span#remoteIpAndPort(String,
-   * int) remote IP and port} from the {@link #unwrap() delegate}. Defaults to false.
+   * Used by {@link HttpServerHandler#handleReceive(HttpServerRequest)} to add remote socket
+   * information about the client from the {@link #unwrap() delegate}.
    *
-   * @see HttpServerAdapter#parseClientIpAndPort(Object, Span)
+   * <p>By default, this tries to parse the {@link #parseClientIpFromXForwardedFor(Span) forwarded
+   * IP}. Override to add client socket information when forwarded info is not available.
+   *
+   * <p>Aside: the ability to parse socket information on server request objects is likely even if
```

Comment:
Not sure I parse this sentence well. Is it the same as It is more likely a server request object will be able to parse socket information as opposed to a client object.

Reply:
thanks for the help!